admin_manual/enterprise/external_storage/s3_swift_as_primary_object_store_configuration.rst

```diff
@@ -1,159 +1,167 @@
+In ownCloud Enterprise edition administrators can configure Amazon S3 objects as the primary storage.
+Doing this replaces the default ownCloud ``owncloud/data`` directory.
+
+You may still need to keep the ``owncloud/data`` directory for these reasons:
 
 * The ownCloud log file is saved in the data directory
 * Legacy apps may not support using anything but the ``owncloud/data`` directory
 
-You can move your logfile by changing its location in ``config.php``. You may still need 
-``owncloud/data`` for backwards compatibility with some apps.
+Administrators can move their ownCloud log file, by changing its location in ``config.php``.
+However, even when this is done, ``owncloud/data`` may still be required for backward compatibility with some apps.
+
+
+`The S3 Object Storage app`_ (``files_primary_s3``) is the preferred way to provide S3 storage support,
+However, `the Object Storage Support app`_ (``objectstore``) is still available.
+
+.. note:: OpenStack Swift has been deprecated.
+
+When using ``files_primary_s3``, the Amazon S3 bucket needs to be created manually according to the developer documentation, and versioning needs to be enabled.
+
+.. note:: ownCloud GmbH provides consulting for migrations from ``objectstore`` to ``files_primary_s3``.
+
+.. note:: The source of the S3 Object Storage app `is available on gitHub`_.
 
 Implications
 ------------
-This configuration needs to be applied before the first login of any user – including the 
-admin user. Otherwise, you will run into an error condition, because ownCloud cannot find 
-the user's files anymore.
 
-ownCloud in object store mode expects exclusive access to the object store 
-container, because it only stores the binary data for each file. The metadata 
-are kept in the local database for performance reasons.
+- Apply this configuration before the first login of any user – including the admin user; otherwise, ownCloud can no longer find the user's files.
+
+- ownCloud, in "object store" mode, expects exclusive access to the object store container, because it only stores the binary data for each file. While in this mode, ownCloud stores the metadata in the local database for performance reasons.
 
-The current implementation is incompatible with any app that uses 
-direct file I/O and circumvents the ownCloud virtual filesystem. That includes 
-Encryption and Gallery. Gallery stores thumbnails directly in the 
-filesystem, and Encryption causes severe overhead because key files need 
-to be fetched in addition to any requested file.
+- The current implementation is incompatible with any app that uses direct file I/O (input/output) as it circumvents the ownCloud virtual filesystem. Two excellent examples are:
+
+  #. **The Encryption app:** It fetches critical files in addition to any requested file, which results in significant overhead.
+  #. **The Gallery app:** It stores thumbnails directly in the filesystem.
```

Comment:
Now I know that encryption and gallery are have issues / incompatible with S3.
I would add a kind of „Important note“ paragraph and highlight the situation about this and what to do in case you have installed these apps. Else this info gets lost easily and you are f* up. Not that critical for gallery but essential for encryption !